### DISCUSSION TITLE
Is this a correct way to use setup_module?

### DISCUSSION THREAD

**Discussion by pjyan** [pytest-dev/pytest#12353] [Q&A] [answered]
sys info: `platform linux -- Python 3.7.9, pytest-7.1.2, pluggy-1.0.0 -- /usr/bin/python3
`

Here I have two files, `conftest.py` and `test.py`

```python
#conftest.py
@pytest.fixture(scope="session", autouse=True)
def setup_module(request):
    print('start conftest')
    yield
    print('stop conftest')
```

```python
#test.py
class Test():
    @classmethod
    def setup_class(cls):
        print('setup class')
    def test_func(self):
        pass
```
after running pytest， here is the output：
```python
test.py::Test::test_func start conftest
setup class
PASSEDstop conftest
```

 `start conftest`  is after ` test.py::Test::test_func`, it seems like the function `setup_module` is running after cheking the `test_func`
In my mind. the output is supposed to be something list that
```python
start conftest
...
test.py::Test::test_func 
setup class
PASSED
...
stop conftest
```

**Comment by RonnyPfannschmidt** [comment, marked as answer]
its not clear what you are asking for, `setup_module`as per xunit would be a module level function

your example has a fixture using that name while being session scoped 

a xunit style setup method would be defined in the module where it is wanted

a fixture doesnt need to use the xunit names

these days using fixtures is recommended over using xunit methods

**Reply by pjyan** [reply]
Hi, thanks a lot.
I find the problem -- mix using fixture and xunit names, which is not necessary
